src/core/util.ts

```diff
@@ -116,6 +116,23 @@ export function newObject<T>(obj: T): T | Record<string, unknown> {
   return result
 }
 
+export function getJson(obj: unknown & { toJSON?: () => Record<string, unknown> }): Record<string, unknown> {
+  if (typeof (obj) !== 'object' || obj === null) {
+    return {}
+  }
+
+  if (typeof obj.toJSON === 'function') {
+    return obj.toJSON()
+  }
+
+  const result = {}
+  for (const k in obj) {
+    result[k] = typeof obj[k] === 'object' ? getJson(obj[k]) : obj[k]
+  }
+
+  return result
```

Comment:
This could be simplified to `JSON.parse(JSON.stringify(obj))`, though I think the current solution is better for performance.

Reply:
I ended up using `JSON.parse(JSON.stringify(obj))` because integration tests where failing and I couldn't figure out why.